src/components/UrqlSubscription.re

```diff
@@ -0,0 +1,64 @@
+[@bs.module "urql"]
+external subscriptionComponent: ReasonReact.reactClass = "Subscription";
+
+type subscriptionRenderPropsJs('a) = {
+  .
+  "fetching": bool,
+  "data": Js.Nullable.t('a),
+  "error": Js.Nullable.t(UrqlCombinedError.t),
+};
+
+type subscriptionRenderProps('a) = {
+  fetching: bool,
+  data: option('a),
+  error: option(UrqlCombinedError.t),
+  response: UrqlTypes.response('a),
+};
+
+let urqlDataToRecord = (result: subscriptionRenderPropsJs('a)) => {
+  let data = result##data |> Js.Nullable.toOption;
+  let error = result##error |> Js.Nullable.toOption;
+
+  let response: UrqlTypes.response('a) =
+    switch (result##fetching, data, error) {
+    | (true, _, _) => Fetching
+    | (false, Some(data), _) => Data(data)
+    | (false, _, Some(error)) => Error(error)
+    | (false, None, None) => NotFound
+    };
+
+  {fetching: result##fetching, data, error, response};
+};
+
+type handler('a, 'b) =
```

Comment:
I did decide to make these named arguments. Also, the `urql` type here: https://github.com/FormidableLabs/urql/blob/master/src/components/Subscription.tsx#L13 specified the type of previous subscriptions as `any | void` (which I think may actually better be written as `any | undefined`). For this reason, we'll wrap this in an option to force users to handle the previous case where `~prevSubscriptions` is `undefined` (which is equivalent to Reason's `None`).